toolbox/src/base/WBBlock.cpp

```diff
@@ -0,0 +1,275 @@
+#include "WBBlock.h"
+
+#include "BlockInformation.h"
+#include "Log.h"
+#include "ToolboxSingleton.h"
+#include "Configuration.h"
+#include "RobotInterface.h"
+#include "AnyType.h"
+#include <string>
+#include <memory>
+#include <iDynTree/Core/MatrixFixSize.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/Twist.h>
+#include <iDynTree/Core/Transform.h>
+
+using namespace wbt;
+
+const unsigned WBBlock::PARAM_IDX_CONFIGURATION = 1; // Struct from Simulink
+const unsigned WBBlock::PARAM_IDX_CONFBLOCKNAME = 2; // Absolute name of the block containing the
+                                                     // configuration
+
+bool WBBlock::initializeRobotState(unsigned dofs)
+{
+    using namespace iDynTree;
+    robotState.m_gravity        = new Vector3(getConfiguration().getGravityVector().data(), 3);
+    robotState.m_jointsPosition = new VectorDynSize(dofs);
+    robotState.m_jointsPosition->zero();
+    robotState.m_world_T_base   = new Transform();
+    robotState.m_baseVelocity   = new Twist();
+    robotState.m_jointsVelocity = new VectorDynSize(dofs);
+    robotState.m_jointsVelocity->zero();
+    return robotState.m_gravity && robotState.m_jointsPosition && robotState.m_world_T_base &&
+           robotState.m_baseVelocity && robotState.m_jointsVelocity;
+}
+
+unsigned WBBlock::numberOfParameters() { return 2; }
+
+bool WBBlock::getWBToolboxParameters(Configuration& config, BlockInformation* blockInfo)
```

Comment:
I probably forgot at the beginning to add `const` specifier to the blockInfo parameter (in the whole toolbox. It does not make sense to modify it once created by Simulink)